x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/utils.ts

```diff
@@ -1027,3 +1050,83 @@ export const getDisabledActionsWarningText = ({
     return `${alertsGeneratedText} connector ${actionTypesJoined} is not enabled. To send notifications, you need a higher Security Analytics license / tier`;
   }
 };
+
+export const isEqlBuildingBlockAlert = (
+  alertObject: unknown
+): alertObject is EqlBuildingBlockAlert800 =>
+  (alertObject as EqlBuildingBlockAlert800)?.[ALERT_BUILDING_BLOCK_TYPE] != null;
+
+export const isEqlShellAlert = (alertObject: unknown): alertObject is EqlShellAlert800 =>
+  (alertObject as EqlShellAlert800)?.[ALERT_UUID] != null &&
+  (alertObject as EqlShellAlert800)?.[ALERT_GROUP_ID] != null;
+
+type RuleWithInMemorySuppression = ThreatRuleParams | EqlRuleParams | MachineLearningRuleParams;
+
+export interface SequenceSuppressionTermsAndFieldsParams {
+  shellAlert: WrappedEqlShellOptionalSubAlertsType;
+  buildingBlockAlerts: Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>>;
+  spaceId: string;
+  completeRule: CompleteRule<RuleWithInMemorySuppression>;
+  mergeStrategy: ConfigType['alertMergeStrategy'];
+  indicesToQuery: string[];
+  buildReasonMessage: BuildReasonMessage;
+  alertTimestampOverride: Date | undefined;
+  ruleExecutionLogger: IRuleExecutionLogForExecutors;
+  publicBaseUrl: string | undefined;
+  primaryTimestamp: string;
+  secondaryTimestamp?: string;
+}
+
+export type SequenceSuppressionTermsAndFieldsFactory = (
+  shellAlert: WrappedEqlShellOptionalSubAlertsType,
+  buildingBlockAlerts: Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>>,
+  buildReasonMessage: BuildReasonMessage
+) => WrappedFieldsLatest<EqlShellFieldsLatest & SuppressionFieldsLatest> & {
+  subAlerts: Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>>;
+};
+
+export const sequenceSuppressionTermsAndFieldsFactory = ({
+  shellAlert,
+  buildingBlockAlerts,
+  spaceId,
+  completeRule,
+  mergeStrategy,
+  indicesToQuery,
+  buildReasonMessage,
+  alertTimestampOverride,
+  ruleExecutionLogger,
+  publicBaseUrl,
+  primaryTimestamp,
+  secondaryTimestamp,
+}: SequenceSuppressionTermsAndFieldsParams): WrappedFieldsLatest<
+  EqlShellFieldsLatest & SuppressionFieldsLatest
+> & {
+  subAlerts: Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>>;
+} => {
+  console.error('ARE WE IN HERE');
+  const suppressionTerms = getSuppressionTerms({
+    alertSuppression: completeRule?.ruleParams?.alertSuppression,
+    fields: shellAlert?._source,
+  });
+  console.error('SUPPRESSION TERMS', suppressionTerms);
+  const instanceId = objectHash([suppressionTerms, completeRule.alertId, spaceId]);
+
+  const suppressionFields = getSuppressionAlertFields({
+    primaryTimestamp,
+    secondaryTimestamp,
+    // as casting should work because the alert fields are flattened (hopefully?)
+    fields: shellAlert?._source as Record<string, string | number | null> | undefined,
+    suppressionTerms,
+    fallbackTimestamp: alertTimestampOverride?.toISOString() ?? new Date().toISOString(),
```

Comment:
`fallbackTimestamp` should be the shellAlert `@timestamp` in line with other usages of `getSuppressionAlertFields`